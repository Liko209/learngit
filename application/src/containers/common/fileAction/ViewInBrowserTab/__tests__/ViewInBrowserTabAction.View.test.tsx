import React from 'react';
import { ViewInBrowserTabAction } from '../ViewInBrowserTabAction.View';
import { shallow } from 'enzyme';
import { Translation } from 'react-i18next';

jest.mock('@/common/isUserAgent', () => ({ isElectron: false }));
describe('ViewInBrowserTabAction', () => {
  const wrapper = shallow(<ViewInBrowserTabAction item={{ name: '' }} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should open a window', done => {
    global.window.open = jest.fn();
    wrapper.simulate('click');

    setTimeout(() => {
      expect(global.window.open).toHaveBeenCalled();
      done();
    });
  });

  it('should show correct text in browser [JPT-2926]', () => {
    expect(wrapper.find(Translation).shallow()).toMatchSnapshot();
  });
});
