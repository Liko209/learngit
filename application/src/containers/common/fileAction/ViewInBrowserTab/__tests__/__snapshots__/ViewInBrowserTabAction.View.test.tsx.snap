// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewInBrowserTabAction should render correctly 1`] = `
<ForwardRef
  data-test-automation-id="viewInBrowserTab"
  icon={
    <ForwardRef(JuiIconography)
      iconColor={
        Array [
          "grey",
          "500",
        ]
      }
      iconSize="small"
    >
      edit
    </ForwardRef(JuiIconography)>
  }
  onClick={[Function]}
>
  <Translation
    ns="translations"
  >
    <Component />
  </Translation>
</ForwardRef>
`;

exports[`ViewInBrowserTabAction should show correct text in browser and electron 1`] = `"message.fileAction.viewInNewTab"`;
